indico/modules/events/papers/client/js/components/CommentForm.jsx

```diff
@@ -0,0 +1,148 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2020 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import React, {useState} from 'react';
+import PropTypes from 'prop-types';
+import {useSelector} from 'react-redux';
+import {Form as FinalForm} from 'react-final-form';
+import {Button, Form} from 'semantic-ui-react';
+
+import {Translate} from 'indico/react/i18n';
+import {FinalDropdown, FinalInput, FinalSubmitButton, FinalTextArea} from 'indico/react/forms';
+
+import {canJudgePaper, canReviewPaper} from '../selectors';
+
+const visibilityOptions = {
+  judges: {
+    value: 'judges',
+    text: Translate.string('Visible only to judges'),
+  },
+  reviewers: {
+    value: 'reviewers',
+    text: Translate.string('Visible to reviewers and judges'),
+  },
+  contributors: {
+    value: 'contributors',
+    text: Translate.string('Visible to contributors, reviewers and judges'),
+  },
+};
+
+/**
+ * @param {Function} onSubmit - function invoked on the form submission
+ * @param {Function} onToggleExpand - function called every time the text input gets/loses focus
+ * @param {Object} comment - object with the data used to prefill the form's initial values
+ * @param {Boolean} expanded - whether the form should be expanded from the start
+ */
+export default function CommentForm({onSubmit, onToggleExpand, comment, expanded}) {
+  const [commentFormVisible, setCommentFormVisible] = useState(expanded);
+  const canReview = useSelector(canReviewPaper);
+  const canJudge = useSelector(canJudgePaper);
+  const InputComponent = commentFormVisible ? FinalTextArea : FinalInput;
+  const inputProps = commentFormVisible
+    ? {autoFocus: true}
+    : {
+        onFocus: () => {
+          setCommentFormVisible(true);
+          onToggleExpand(true);
+        },
+      };
+
+  const handleSubmit = async (formData, form) => {
+    const error = await onSubmit(formData, form);
+    if (error) {
+      return error;
+    }
+
+    setCommentFormVisible(false);
+    if (onToggleExpand) {
+      onToggleExpand(false);
+    }
+
+    setTimeout(() => form.reset(), 0);
+  };
+
+  const initialValues = comment ? {comment: comment.text} : {};
+  if (!canJudge) {
+    delete visibilityOptions.judges;
```

Comment:
this is still modifying something defined outside this function..

Reply:
Ah.. ok. I think I changed it somewhere else :P